src/Compilers/VisualBasic/Test/Semantic/Semantics/OverloadResolutionPriorityTests.vb

```diff
@@ -0,0 +1,987 @@
+
+            Dim comp2 = CreateCompilation(source, references:={comp1.ToMetadataReference()}, options:=TestOptions.DebugExe)
+            CompileAndVerify(comp2, expectedOutput:="1").VerifyDiagnostics()
+
+            Dim comp3 = CreateCompilation(source, references:={comp1.EmitToImageReference()}, options:=TestOptions.DebugExe)
+            CompileAndVerify(comp3, expectedOutput:="1").VerifyDiagnostics()
+        End Sub
+
+        <Fact>
+        Public Sub ParameterlessProperty_01()
+            Dim compilationDef = "
+Module Module1
+
+    Sub Main()
+        M1 = 0
+    End Sub
+
+    <System.Runtime.CompilerServices.OverloadResolutionPriority(-1)>
+    WriteOnly Property M1 As Integer
+        Set
+            System.Console.Write(1)
+        End Set
+    End Property
+
+    WriteOnly Property M1(Optional x As Integer = 0) As Integer
+        Set
+            System.Console.Write(2)
+        End Set
+    End Property
+End Module
+"
+            Dim compilation = CreateCompilation({compilationDef, OverloadResolutionPriorityAttributeDefinitionVB}, options:=TestOptions.ReleaseExe)
+            CompileAndVerify(compilation, expectedOutput:="2")
+        End Sub
+
+        <Fact>
+        Public Sub ParameterlessProperty_02()
+            Dim compilationDef = "
+Module Module1
+
+    Sub Main()
+        M1 = 0
+    End Sub
+
+    WriteOnly Property M1 As Integer
+        Set
+            System.Console.Write(1)
+        End Set
+    End Property
+
+    <System.Runtime.CompilerServices.OverloadResolutionPriority(1)>
+    WriteOnly Property M1(Optional x As Integer = 0) As Integer
+        Set
+            System.Console.Write(2)
+        End Set
+    End Property
+End Module
+"
+            Dim compilation = CreateCompilation({compilationDef, OverloadResolutionPriorityAttributeDefinitionVB}, options:=TestOptions.ReleaseExe)
+            CompileAndVerify(compilation, expectedOutput:="2")
+        End Sub
+
+        <Fact>
+        Public Sub DefaultProperty_01()
+            Dim compilationDef = "
+Class Module1
+
+    Shared Sub Main()
+    End Sub
+
+    <System.Runtime.CompilerServices.OverloadResolutionPriority(-1)>
+    Default WriteOnly Property M1 As Integer
+        Set
+            System.Console.Write(1)
+        End Set
+    End Property
+
+    Default WriteOnly Property M1(x As Integer) As Integer
+        Set
+            System.Console.Write(2)
+        End Set
+    End Property
+End Class
+"
+            Dim compilation = CreateCompilation({compilationDef, OverloadResolutionPriorityAttributeDefinitionVB}, options:=TestOptions.ReleaseExe)
+            compilation.AssertTheseDiagnostics(
+<expected>
+BC31048: Properties with no required parameters cannot be declared 'Default'.
+    Default WriteOnly Property M1 As Integer
+                               ~~
+</expected>)
+        End Sub
+
+        <Fact>
+        Public Sub DefaultProperty_02()
```

Comment:
>DefaultProperty_02 [](http://example.com/codeflow?start=19&length=18)

Is this test affected by the compiler change?

Reply:
> Is this test affected by the compiler change?

No. But the test demonstrates that there are no interesting scenarios when a parameter-less property is overloaded with default property